src/ol/css.js

```diff
@@ -114,3 +114,11 @@ export const getFontParameters = function (fontSpec) {
   style.families = style.family.split(/,\s?/);
   return style;
 };
+
+/**
+ * @param {number} opacity Opacity (0..1).
+ * @return {string} CSS opacity.
+ */
+export function cssOpacity(opacity) {
+  return opacity === 1 ? '' : String(opacity);
+}
```

Comment:
The css opacity may be rounded when it is set on a html node. E. g. `1 / 3` becomes `"0.333333"`.
To allow container reuse in that case it might make sense to do something similar to the toString transform function. Lines 285-288
https://github.com/openlayers/openlayers/blob/cb6995d71ae84bfa41e93852d23dae5019814b0b/src/ol/transform.js#L274-L289

Reply:
I decided to use simple rounding instead of going through the DOM, because this function would be called with every render frame, and the required DOM operation is probably almost as expensive as setting the opacity on every render frame, which I wanted to avoid.